src/main/resources/config/liquibase/changelog/20230920133000_changelog.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+
+    <changeSet author="bassner" id="20230920133000">
+        <!-- Create exam_live_event table -->
+        <createTable tableName="exam_live_event">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="created_by" type="varchar(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="created_date" type="timestamp">
+                <constraints nullable="false"/>
+            </column>
+            <column name="exam_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="student_exam_id" type="bigint">
+                <constraints nullable="true"/>
+            </column>
+            <column name="discriminator" type="varchar(1)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="text_content" type="text"/>
+            <column name="old_working_time" type="int"/>
+            <column name="new_working_time" type="int"/>
+            <column name="course_wide" type="boolean"/>
+        </createTable>
+
+        <!-- Add foreign keys with delete cascade -->
+        <addForeignKeyConstraint baseTableName="exam_live_event" baseColumnNames="exam_id"
+                                 constraintName="fk_exam_live_event_exam_id"
+                                 referencedTableName="exam" referencedColumnNames="id"
+                                 onDelete="CASCADE"/>
```

Comment:
Is this correct? If I interpret this correctly, it would mean that we would delete the exam if the event is deleted

Reply:
No, it's the other way around. In pure SQL, this would be:

```sql
ALTER TABLE exam_live_event
ADD CONSTRAINT fk_exam_live_event_exam_id
    FOREIGN KEY (exam_id) REFERENCES exam(id)
    ON DELETE CASCADE;
```

which means that we add a reference from the `exam_live_event` table to the `exam` table. If we try to delete an exam with an existing live event, we would violate this constraint as the stored `exam_id` must point to an existing exam. Per default, the DB would then reject the operation (`ON DELETE REJECT`); here, we tell it to also delete the live event with the exam. This allows us to just delete the exam without having to manually send a delete query for the events beforehand.

The exam table does not have a reference to the live events table; delete operations on live events will therefore not affect the exam table.